service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/TaskHandlingTest.java

```diff
@@ -0,0 +1,248 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.gp2gp.common.task.TaskConsumer;
+import uk.nhs.adaptors.gp2gp.common.task.TaskDispatcher;
+import uk.nhs.adaptors.gp2gp.common.task.TaskExecutor;
+import uk.nhs.adaptors.gp2gp.common.task.TaskExecutorFactory;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.testcontainers.ActiveMQExtension;
+import uk.nhs.adaptors.gp2gp.testcontainers.MongoDBExtension;
+
+@RunWith(SpringRunner.class)
+@ExtendWith({SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class, MockitoExtension.class})
+@SpringBootTest
+@DirtiesContext
+public class TaskHandlingTest {
+
+    private static final String TASK_TYPE_HEADER_NAME = "TaskType";
+    private static final String NACK_TYPE_CODE = "AE";
+
+    @SpyBean
+    private ObjectMapper objectMapper;
+    @Autowired
+    private TaskConsumer taskConsumer;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+    @SpyBean
+    private EhrExtractStatusService ehrExtractStatusService;
+    @SpyBean
+    private TaskExecutorFactory taskExecutorFactory;
+    @MockBean
+    private TaskDispatcher taskDispatcher;
+    @Mock
+    private Message message;
+    @Mock
+    private TaskExecutor taskExecutor;
+
+
+    @BeforeEach
+    public void setup() {
+        lenient().doReturn(taskExecutor).when(taskExecutorFactory).getTaskExecutor(any());
+    }
+
+    @Test
+    @SneakyThrows
+    public void When_MessageIsUnreadable_Expect_MessageNotToBeProcessed() {
+        when(message.getStringProperty(TASK_TYPE_HEADER_NAME)).thenReturn(SendEhrExtractCoreTaskDefinition.class.getName());
+        when(message.getBody(any())).thenReturn("not a valid body");
+
+        taskConsumer.receive(message);
+
+        verify(message, never()).acknowledge();
+        verifyNoInteractions(taskExecutorFactory, ehrExtractStatusService);
+    }
+
+    @Test
+    @SneakyThrows
+    public void When_NonNackTaskProcessingFails_Expect_WholeProcessToBeFailed() {
+        // ARRANGE
+        mockSendEhExtractCoreTaskMessage();
+        doThrow(RuntimeException.class).when(taskExecutor).execute(any());
+        var ehrExtractStatus = createEhrExtractStatusInDb(false);
+
+        // ACT
```

Comment:
do we need these comments? (optional approve otherwise

Reply:
They're not necessary, but commonly used to separate the sections of each test. If you're not happy with them, I can remove them.

Reply:
Yeah I agree they aren't necessary

Reply:
removed